test/response/original-response.test.ts

```diff
@@ -0,0 +1,61 @@
+import http from 'http'
+import { RequestInterceptor } from '../../src'
+
+function createServer(port = 1337, host = 'localhost'): Promise<http.Server> {
+  const server = http.createServer((_, res) => {
+    res.writeHead(200)
+    res.end('OK')
+  })
+  server.on('connection', (socket) => {
+    socket.on('close', () => {
+      // Special event that we can use to make sure that we are assering when
+      // server is done with the response
+      server.emit('connection-end')
+    })
+  })
+  return new Promise((resolve) => {
+    server.listen(port, host, () => resolve(server))
+  })
+}
+
+function intercept() {
+  let interceptor: RequestInterceptor | null = new RequestInterceptor()
+  interceptor.use(() => {
```

Comment:
Do you think we should still assert that the middleware function (the one supplied to `interceptor.use()` is called?